main/src/ca/uwaterloo/flix/language/phase/unification/Unification.scala

```diff
@@ -70,21 +69,41 @@ object Unification {
   def unifyTypes(tpe1: Type, tpe2: Type, renv: RigidityEnv)(implicit flix: Flix): Result[Substitution, UnificationError] = {
     (tpe1, tpe2) match {
 
-      // don't try to unify effects if the `no-set-effects` flag is on
-      case (x, y) if x.kind == Kind.Effect && y.kind == Kind.Effect && flix.options.xnoseteffects => Ok(Substitution.empty)
-
-      case (x: Type.Var, y: Type.Var) => unifyVars(x.asKinded, y.asKinded, renv)
+      //
+      // Effects
+      //
+      case _ if tpe1.kind == Kind.Effect && tpe2.kind == Kind.Effect =>
```

Comment:
We could probably just match on the kinds directly know, right? Instead of this `_ =>`.

Reply:
Yes